packages/core/src/selectors/base.ts

```diff
@@ -0,0 +1,45 @@
+import { QueryBundle, ToolMetadataOnlyDescription } from "../types";
+
+export interface SingleSelection {
+  index: number;
+  reason: string;
+}
+
+export type SelectorResult = {
+  selections: SingleSelection[];
+};
+
+type QueryType = string | QueryBundle;
+
+function wrapChoice(
+  choice: string | ToolMetadataOnlyDescription,
```

Comment:
Why do we want this ToolMetadataOnlyDescription? This should be able to take in any ToolMetadata.

Reply:
@yisding @marcusschiesser suggestion was to have only a description since we aren't using the name on ToolMetadata, and I agree because makes the use easier and we don't need to make name optional for other engines that requires it